base/pointer.jl

```diff
@@ -159,3 +159,6 @@ isless(x::Ptr{T}, y::Ptr{T}) where {T} = x < y
 +(x::Ptr, y::Integer) = oftype(x, add_ptr(UInt(x), (y % UInt) % UInt))
 -(x::Ptr, y::Integer) = oftype(x, sub_ptr(UInt(x), (y % UInt) % UInt))
 +(x::Integer, y::Ptr) = y + x
+
+unsigned(x::Ptr) = convert(Unsigned, x)
+signed(x::Ptr) = convert(Signed, x)
```

Comment:
```suggestion
signed(x::Ptr) = Int(x)
```

Reply:
Does this cause any difference? Also, do we need `unsigned(x::Ptr) = UInt(x)` as well?

Reply:
I think this can happen:
```
julia> convert(Signed, Ptr{Cvoid}(0xffffffffffffffff))
ERROR: InexactError: check_top_bit(Int64, 18446744073709551615)
```
`UInt(x)` works too, but either works in the Unsigned case.

Reply:
Exactly! Thank you for your explanation. I've updated both for symmetry.